roles/oneagent/tests/conmponent/run.py

```diff
@@ -0,0 +1,166 @@
+    return subprocess.Popen(["python", server_path], env=get_env_vars(),
+                            stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding="utf-8")
+
+
+def get_file_content(path: Path):
+    with open(path, "r") as f:
+        return f.readlines()
+
+
+def replace_tag(source: List[str], old: str, new: str) -> List[str]:
+    return [line.replace(old, new) for line in source]
+
+
+def prepare_installers():
+    logging.info("Preparing installers...")
+
+    oneagentctl_bin_name = "oneagentctl.sh"
+    uninstall_script_name = "uninstall.sh"
+    installer_partial_name = "Dynatrace-OneAgent-Linux"
+
+    version_tag = "##VERSION##"
+    uninstall_code_tag = "##UNINSTALL_CODE##"
+    oneagentctl_code_tag = "##ONEAGENTCTL_CODE##"
+
+    resource_dir = Path("resources") / "installers"
+
+    uninstall_template = get_file_content(resource_dir / uninstall_script_name)
+    uninstall_code = replace_tag(uninstall_template, "$", r"\$")
+
+    oneagentctl_template = get_file_content(resource_dir / oneagentctl_bin_name)
+    oneagentctl_code = replace_tag(oneagentctl_template, "$", r"\$")
+
+    installer_template = get_file_content(resource_dir / f"{installer_partial_name}.sh")
+    installer_template = replace_tag(installer_template, uninstall_code_tag, "".join(uninstall_code))
+    installer_template = replace_tag(installer_template, oneagentctl_code_tag, "".join(oneagentctl_code))
+
+    timestamp = '{:%Y%m%d-%H%M%S}'.format(datetime.now())
+    # Minimal supported version is 1.199
+    for version in ["1.199.0", "1.300.0"]:
+        full_version = f"{version}.{timestamp}"
+        installer_code = replace_tag(installer_template, version_tag, full_version)
+        with open(INSTALLERS_DIR / f"{installer_partial_name}-{full_version}.sh", "w") as f:
+            f.writelines(installer_code)
+
+
+def prepare_environment():
+    logging.basicConfig(
+        format="%(asctime)s [server] %(levelname)s: %(message)s", datefmt="%H:%M:%S", level=logging.INFO
+    )
+    shutil.rmtree(TEST_DIR, ignore_errors=True)
+    os.makedirs(INSTALLERS_DIR, exist_ok=True)
+    os.makedirs(LOG_DIR, exist_ok=True)
+    prepare_installers()
+
+
+def parse_args() -> Dict[str, Any]:
+    parser = argparse.ArgumentParser(
+        description="Run component tests for OneAgent role. If any of specified platform contains"
+                    " 'localhost' as an IP address, the script will perform tests on the local machine."
+                    " In such case, user and password arguments can be empty"
+    )
+
+    parser.add_argument(f"--{USER_KEY}", type=str, help="User for remote hosts", default="")
+    parser.add_argument(f"--{PASS_KEY}", type=str, help="Password for remote hosts user", default="")
+
+    for platform in DeploymentPlatform:
+        parser.add_argument(
+            f"--{platform.value}", type=str, nargs="+", default=[], help="List of IPs for specified platform"
+        )
+
+    return vars(parser.parse_args())
+
+
+def main():
```

Comment:
It would be better to run the tests directly using `pytest`, but I understand that it would take a lot of work to refactor the code.